### PR TITLE
feat: Adds atlas Organization New Parameters Support 

### PR DESCRIPTION
**Pull request by maastha**
## Description

- Adds support for new setting params to organization resource
- Adds validation to create handler to ensure role_names for new API key contains ORG_OWNER permissions.
- Updates documentation & example
- Updates acceptance tests

Link to any related issue(s): https://jira.mongodb.org/browse/CLOUDP-219771

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [x] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

## Further comments

```hcl
main.tf:

resource "mongodbatlas_organization" "test" {
	org_owner_id = "......"
	name = "manual-org-renamed"
	description = "temp desc"
	role_names = ["ORG_OWNER", "ORG_BILLING_ADMIN"]
        multi_factor_auth_required = false
        api_access_list_required = false
        restrict_employee_access = true
	  }

```
```json
tfstate:

  "resources": [
    {
      "mode": "managed",
      "type": "mongodbatlas_organization",
      "name": "test",
      "provider": "provider[\"registry.terraform.io/mongodb/mongodbatlas\"]",
      "instances": [
        {
          "schema_version": 0,
          "attributes": {
            "api_access_list_required": false,
            "description": "temp desc",
            "federation_settings_id": null,
            "id": "b3JnX2lk:NjVhMWIyZjAwNTJjMzc3ZTY5MDk4Yzlj",
            "multi_factor_auth_required": false,
            "name": "manual-org-renamed",
            "org_id": ".........",
            "org_owner_id": "........",
            "private_key": "..............",
            "public_key": ".........",
            "restrict_employee_access": true,
            "role_names": [
              "ORG_BILLING_ADMIN",
              "ORG_OWNER"
            ]
          },
          "sensitive_attributes": [],
          "private": "bnVsbA=="
        }
      ]
    }
  ]
```
image

### CODE REVIEW

File: internal/service/organization/data_source_organization_test.go [hunk 1]
```diff
@@ -6,21 +6,26 @@ import (
...
 		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
 		Steps: []resource.TestStep{
 			{
 				Config: testAccMongoDBAtlasOrganizationConfigWithDS(orgID),
 				Check: resource.ComposeTestCheckFunc(
-					resource.TestCheckResourceAttrSet("data.mongodbatlas_organization.test", "name"),
-					resource.TestCheckResourceAttrSet("data.mongodbatlas_organization.test", "id"),
+					resource.TestCheckResourceAttrSet(datasourceName, "name"),
+					resource.TestCheckResourceAttrSet(datasourceName, "id"),
+					resource.TestCheckResourceAttrSet(datasourceName, "restrict_employee_access"),
```
Comment (lantoli) [inline internal/service/organization/data_source_organization_test.go:26]: shouldn't we also check for the expected value?

File: internal/service/organization/resource_organization.go [hunk 2]
```diff
@@ -74,7 +93,30 @@ func resourceMongoDBAtlasOrganizationCreate(ctx context.Context, d *schema.Resou
...
+	conn = clients.(*config.MongoDBClient).AtlasV2
+
+	_, _, err = conn.OrganizationsApi.UpdateOrganizationSettings(ctx, orgID, newOrganizationSettings(d)).Execute()
+	if err != nil {
+		if _, _, err := conn.OrganizationsApi.DeleteOrganization(ctx, orgID).Execute(); err != nil {
+			d.SetId("")
+			return diag.FromErr(fmt.Errorf("an error occurred when updating Organization settings. Unable to delete organization, there may be dangling resources: %s", err))
+		}
+		d.SetId("")
+		return diag.FromErr(fmt.Errorf("error setting Organization setings: %s", err))
```
Comment (marcosuma) [inline internal/service/organization/resource_organization.go]:
```suggestion
		return diag.FromErr(fmt.Errorf("an error occurred when updating Organization settings: %s", err))
```

File: internal/service/organization/resource_organization.go [hunk 3]
```diff
@@ -74,7 +93,30 @@ func resourceMongoDBAtlasOrganizationCreate(ctx context.Context, d *schema.Resou
...
+	}
+
+	clients, _ := cfg.NewClient(ctx)
+	conn = clients.(*config.MongoDBClient).AtlasV2
+
+	_, _, err = conn.OrganizationsApi.UpdateOrganizationSettings(ctx, orgID, newOrganizationSettings(d)).Execute()
+	if err != nil {
+		if _, _, err := conn.OrganizationsApi.DeleteOrganization(ctx, orgID).Execute(); err != nil {
+			d.SetId("")
+			return diag.FromErr(fmt.Errorf("an error occurred when updating Organization settings. Unable to delete organization, there may be dangling resources: %s", err))
```
Comment (marcosuma) [inline internal/service/organization/resource_organization.go]: is it possible here to return both errors?

File: internal/service/organization/resource_organization.go [hunk 4]
```diff
@@ -185,3 +254,21 @@ func newCreateOrganizationRequest(d *schema.ResourceData) *admin.CreateOrganizat
...
+	return &admin.OrganizationSettings{
+		ApiAccessListRequired:   pointy.Bool(d.Get("api_access_list_required").(bool)),
+		MultiFactorAuthRequired: pointy.Bool(d.Get("multi_factor_auth_required").(bool)),
+		RestrictEmployeeAccess:  pointy.Bool(d.Get("restrict_employee_access").(bool)),
+	}
+}
+
+func isAPIKeyOrgOwner(roles []string) bool {
+	for _, role := range roles {
+		if role == "ORG_OWNER" {
```
Comment (marcosuma) [inline internal/service/organization/resource_organization.go]: should we have ORG_OWNER as const?

File: internal/service/organization/resource_organization.go [hunk 5]
```diff
@@ -61,11 +61,30 @@ func Resource() *schema.Resource {
...
+				Optional: true,
+				Computed: true,
+			},
 		},
 	}
 }
 
 func resourceMongoDBAtlasOrganizationCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	if !isAPIKeyOrgOwner(conversion.ExpandStringList(d.Get("role_names").(*schema.Set).List())) {
+		return diag.FromErr(fmt.Errorf("`role_names` for new API Key must have the ORG_OWNER role to use this resource"))
```
Comment (marcosuma) [inline internal/service/organization/resource_organization.go]: nit: you could return the err from the method

File: internal/service/organization/data_source_organization_test.go (hunk 1 above)
Comment (maastha) [reply internal/service/organization/data_source_organization_test.go:26]: updated  for settings, can't check for other attributes as ORG_ID may vary

File: internal/service/organization/resource_organization.go (hunk 2 above)
Comment (maastha) [reply internal/service/organization/resource_organization.go]: updated

File: internal/service/organization/resource_organization.go (hunk 4 above)
Comment (maastha) [reply internal/service/organization/resource_organization.go]: updated

File: internal/service/organization/resource_organization.go (hunk 5 above)
Comment (maastha) [reply internal/service/organization/resource_organization.go]: updated

File: internal/service/organization/resource_organization.go (hunk 3 above)
Comment (maastha) [reply internal/service/organization/resource_organization.go]: updated

**Review by marcosuma — APPROVED** [review]

File: internal/service/organization/resource_organization_test.go [hunk 6]
```diff
@@ -37,17 +40,92 @@ func TestAccConfigRSOrganization_Basic(t *testing.T) {
...
+					resource.TestCheckResourceAttr(resourceName, "restrict_employee_access", "false"),
+					resource.TestCheckResourceAttr(resourceName, "multi_factor_auth_required", "false"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccConfigRSOrganization_BasicAccess(t *testing.T) {
+	acc.SkipTestForCI(t)
```
Comment (lantoli) [inline internal/service/organization/resource_organization_test.go:65]: is it hard to run it in CI / GH?

**Review by lantoli — APPROVED** [review]

**Review by Zuhairahmed — APPROVED** [review]

**Review by AgustinBettati — APPROVED** [review]
LGTM

File: internal/service/organization/resource_organization_test.go (hunk 6 above)
Comment (maastha) [reply internal/service/organization/resource_organization_test.go:65]: sorry this was missed, this requires setting a up a paying organization in cloud-dev that new organizations can be linked to, I couldn't get it to work with my personal organization in cloud-dev (even though I have billing enabled) so requires looking into setting up payments correctly

**Pull request merged**